apps/studio/src/components/PageEditor/LinkEditorModal.tsx

```diff
@@ -102,7 +62,15 @@ const LinkEditorModalContent = ({
     onSave(linkText, linkHref),
   )
 
-  const { siteId } = useQueryParse(editPageSchema)
+  // hacky solution to allow setting the correct state of "Add link" CTA button
+  // while not showing error message on initial render for better UX
+  const [isFirstRender, setIsFirstRender] = useState(true)
+  const showLinkTextErrorState: boolean =
+    !isFirstRender && !!errors.linkText?.message
+
+  useEffect(() => {
+    void trigger()
+  }, [trigger])
```

Comment:
nit: `useLayoutEffect` - this makes it so that this function before repaint and is less likely to mess with other stateful stuff